test/integration/other/select-many.js

```diff
@@ -83,5 +83,72 @@ describe('select many value processing', () => {
             { formId, submissionDefId: one2, path: '/g1/q2', value: 'xyz' }
           ]);
         }))));
+
+  it('should handle when field changes from select1 to select (selectMultiple)', testService(async (service, container) => {
```

Comment:
Yay, thanks for writing this test! I'm happy it passes. I forgot that I think we handle unprocessed values on the fly.